subprojects/analysis-kotlin-descriptors/compiler/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/descriptors/compiler/translator/DefaultDescriptorToDocumentableTranslator.kt

```diff
@@ -1147,14 +1147,14 @@ private class DokkaDescriptorVisitor(
             fullEnumEntryName(),
             DRI(enumClassId.packageFqName.asString(), fullEnumEntryName())
         )
-        is ConstantsKtClassValue -> when (value) {
-            is NormalClass -> (value as NormalClass).value.classId.let {
+        is ConstantsKtClassValue -> when (val value = value) {
```

Comment:
to  avoid https://youtrack.jetbrains.com/issue/KT-22517/Deprecate-smartcasts-for-local-delegated-properties